gdal-sys/build.rs

```diff
@@ -114,41 +119,109 @@ fn main() {
     let mut version = env::var_os("GDAL_VERSION")
         .map(|vs| vs.to_string_lossy().to_string())
         .and_then(|vs| Version::parse(vs.trim()).ok());
+    let use_vcpkg = if cfg!(windows) {
+        env::var_os("GDAL_VCPKG").is_some()
+    } else {
+        false
+    };
 
     let mut found = false;
     if cfg!(windows) {
-        // first, look for a static library in $GDAL_LIB_DIR or $GDAL_HOME/lib
-        // works in windows-msvc and windows-gnu
-        if let Some(ref lib_dir) = lib_dir {
-            let lib_path = lib_dir.join("gdal_i.lib");
-            if lib_path.exists() {
-                prefer_static = true;
-                lib_name = String::from("gdal_i");
-                found = true;
+        if use_vcpkg {
+            let vcpkg_root = env_dir("VCPKG_ROOT");
+            let vcpkg_triplet = env::var("GDAL_VCPKG_TRIPLET");
+
+            if vcpkg_root.is_none() {
+                panic!("GDAL_VCPKG requires VCPKG_ROOT to be set.");
             }
-        }
-        if !found {
-            if let Some(ref home_dir) = home_dir {
-                let home_lib_dir = home_dir.join("lib");
-                let lib_path = home_lib_dir.join("gdal_i.lib");
+
+            if vcpkg_triplet.is_err() {
+                panic!("GDAL_VCPKG requires GDAL_VCPKG_TRIPLET to be set.");
+            }
+
+            let vcpkg_root = vcpkg_root.unwrap();
+            let vcpkg_triplet = vcpkg_triplet.unwrap();
+            prefer_static = vcpkg_triplet.ends_with("-static");
+
+            let vcpkg_install_dir = vcpkg_root.join("installed").join(vcpkg_triplet.clone());
+
+            let pkg_config = env::var("PKG_CONFIG");
+            let pkg_config_path = env::var("PKG_CONFIG_PATH");
+
+            let required_pkg_config = vcpkg_install_dir
+                .join("tools")
+                .join("pkgconf")
+                .join("pkgconf.exe")
+                .to_str()
+                .unwrap()
+                .to_owned();
+            let required_pkg_config_path = vcpkg_install_dir
+                .join("lib")
+                .join("pkgconfig")
+                .to_str()
+                .unwrap()
+                .to_owned();
+
+            let valid_pkg_config = match pkg_config {
```

Comment:
I think this would look a bit nicer like:

```rust
            if !matches!(pkg_config, Some(required_pkg_config)) {
                panic!("GDAL_VCPKG requires PKG_CONFIG to be set to '{required_pkg_config}'.");
            }
```